packages/toolpad-app/src/runtime/ToolpadApp.tsx

```diff
@@ -353,20 +380,37 @@ function RenderedNodeContent({ node, childNodeGroups, Component }: RenderedNodeC
     previousProps.current = props;
   }, [props, argTypes, nodeId, setControlledBinding]);
 
-  // Wrap with slots
+  // Wrap element props
   for (const [propName, argType] of Object.entries(argTypes)) {
-    if (argType?.typeDef.type === 'element') {
+    const isElement = argType?.typeDef.type === 'element';
+    const isElementIterator = argType?.typeDef.type === 'elementIterator';
+
+    if (isElement || isElementIterator) {
+      const value = props[propName];
+
+      let wrappedValue = value;
       if (argType.control?.type === 'slots') {
-        const value = props[propName];
-        props[propName] = <Slots prop={propName}>{value}</Slots>;
+        wrappedValue = <Slots prop={propName}>{value}</Slots>;
       } else if (argType.control?.type === 'slot' || argType.control?.type === 'layoutSlot') {
-        const value = props[propName];
-        props[propName] = (
+        wrappedValue = (
           <Placeholder prop={propName} hasLayout={argType.control?.type === 'layoutSlot'}>
             {value}
           </Placeholder>
         );
       }
+
+      props[propName] = isElementIterator
```

Comment:
Wrap element iterator slot children by iterating and creating a context for the active item.